pkg/controller/razeedeployment/razeedeployment_controller.go

```diff
@@ -856,275 +871,308 @@ func (r *ReconcileRazeeDeployment) Reconcile(request reconcile.Request) (reconci
-		if err != nil {
-			reqLogger.Error(err, "Failed to update JobState")
-			return reconcile.Result{}, err
-		}
-		reqLogger.V(0).Info("Updated JobState")
 		return reconcile.Result{Requeue: true}, nil
-	}
 
-	if foundJob.Status.Succeeded != 1 {
-		reqLogger.Info("waiting for job to finish")
-		return reconcile.Result{RequeueAfter: time.Second * 15}, nil
+	} else if err != nil {
+		reqLogger.Error(err, "Failed to get RemoteResourceS3 from Cluster")
+		return reconcile.Result{}, err
 	}
 
-	// if the job succeeds apply the parentRRS3 and patch the Infrastructure, ClusterVersion and Console resources
-	if foundJob.Status.Succeeded == 1 {
-		parentRRS3 := &unstructured.Unstructured{}
-		parentRRS3.SetGroupVersionKind(schema.GroupVersionKind{
-			Group:   "deploy.razee.io",
-			Kind:    "RemoteResourceS3",
-			Version: "v1alpha2",
-		})
+	if err := controllerutil.SetControllerReference(instance, watchKeeperDeployment, r.scheme); err != nil {
+		reqLogger.Error(err, "Failed to set controller reference")
+		return reconcile.Result{}, err
+	}
 
-		err = r.client.Get(context.TODO(), client.ObjectKey{Name: utils.PARENT_RRS3_RESOURCE_NAME, Namespace: *instance.Spec.TargetNamespace}, parentRRS3)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				reqLogger.V(0).Info("Resource does not exist", "resource: ", utils.PARENT_RRS3)
-				parentRRS3 := r.makeParentRemoteResourceS3(instance)
-
-				err = r.client.Create(context.TODO(), parentRRS3)
-				if err != nil {
-					reqLogger.Info("Failed to create resource", "resource: ", utils.PARENT_RRS3)
-					return reconcile.Result{}, err
-				}
-				message := "ParentRRS3 install finished"
-				instance.Status.Conditions.SetCondition(status.Condition{
-					Type:    marketplacev1alpha1.ConditionInstalling,
-					Status:  corev1.ConditionTrue,
-					Reason:  marketplacev1alpha1.ReasonParentRRS3Installed,
-					Message: message,
-				})
-
-				_ = r.client.Status().Update(context.TODO(), instance)
-
-				reqLogger.Info("Resource created successfully", "resource: ", utils.PARENT_RRS3)
-				return reconcile.Result{Requeue: true}, nil
-			} else {
-				reqLogger.Info("Failed to get resource", "resource: ", utils.PARENT_RRS3)
-				return reconcile.Result{}, err
-			}
-		}
-		if err == nil {
-			reqLogger.V(0).Info("Resource already exists", "resource: ", utils.PARENT_RRS3)
+	message = "watch-keeper install finished"
+	instance.Status.Conditions.SetCondition(status.Condition{
+		Type:    marketplacev1alpha1.ConditionInstalling,
+		Status:  corev1.ConditionTrue,
+		Reason:  marketplacev1alpha1.ReasonWatchKeeperDeploymentInstalled,
+		Message: message,
+	})
 
-			newParentValues := r.makeParentRemoteResourceS3(instance)
-			updatedParentRRS3 := parentRRS3.DeepCopy()
-			updatedParentRRS3.Object["spec"] = newParentValues.Object["spec"]
+	podList := &corev1.PodList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(*instance.Spec.TargetNamespace),
+		client.MatchingLabels(map[string]string{
+			"owned-by": "rhm",
```

Comment:
This label is not bad, but may be too generic. I like the idea. Let's instead make it "marketplace.redhat.com/app=razee"